boards/components/src/led_matrix.rs

```diff
@@ -75,14 +75,14 @@ macro_rules! led_matrix_component_helper {
 
         const NUM_COLS: usize = count_expressions!($($C),+);
         const NUM_ROWS: usize = count_expressions!($($R),+);
-        static mut BUFFER: [u8; NUM_COLS*NUM_ROWS / 8 + 1] = [0; NUM_COLS*NUM_ROWS / 8 + 1];
+        static mut BUFFER: [u8; NUM_ROWS*NUM_COLS / 8 + 1] = [0; NUM_ROWS*NUM_COLS / 8 + 1];
         components::led_matrix::LedMatrixComponent::new ($alarm_mux, $crate::led_line_component_helper!($Pin, $($C,)+), $crate::led_line_component_helper!($Pin, $($R,)+), $col_active, $row_active, $refresh_rate, &mut BUFFER)
     };};
 }
 
 #[macro_export]
 macro_rules! led_line_component_helper {
-    ($Pin:ty, $($L:expr),+ $(,)?) => {{
+    ($Pin:ty, $($L:expr),+, $(,)?) => {{
```

Comment:
Merge this with the `led_matrix_component_buf`, as that one computes all the constant values.